crates/astria-core/src/protocol/transaction/v1alpha1/action_group/mod.rs

```diff
@@ -0,0 +1,196 @@
+        BridgeUnlockAction,
+        FeeAssetChangeAction,
+        FeeChangeAction,
+        IbcRelayerChangeAction,
+        IbcSudoChangeAction,
+        Ics20Withdrawal,
+        InitBridgeAccountAction,
+        SequenceAction,
+        SudoAddressChangeAction,
+        TransferAction,
+        ValidatorUpdate,
+    },
+    Action,
+};
+
+trait BelongsToGroup {
+    const GROUP: ActionGroup;
+}
+
+macro_rules! impl_belong_to_group {
+    ($(($act:ty, $group:expr)),*$(,)?) => {
+        $(
+            impl BelongsToGroup for $act {
+                const GROUP: ActionGroup = $group;
+            }
+        )*
+    }
+}
+
+impl_belong_to_group!(
+    (SequenceAction, ActionGroup::General),
+    (TransferAction, ActionGroup::General),
+    (ValidatorUpdate, ActionGroup::General),
+    (SudoAddressChangeAction, ActionGroup::UnbundleableSudo),
+    (IbcRelayerChangeAction, ActionGroup::Sudo),
+    (Ics20Withdrawal, ActionGroup::General),
+    (InitBridgeAccountAction, ActionGroup::UnbundleableGeneral),
+    (BridgeLockAction, ActionGroup::General),
+    (BridgeUnlockAction, ActionGroup::General),
+    (BridgeSudoChangeAction, ActionGroup::UnbundleableGeneral),
+    (FeeChangeAction, ActionGroup::Sudo),
+    (FeeAssetChangeAction, ActionGroup::Sudo),
+    (IbcRelay, ActionGroup::General),
+    (IbcSudoChangeAction, ActionGroup::UnbundleableSudo),
+);
+
+impl Action {
+    const fn group(&self) -> ActionGroup {
+        match self {
+            Action::Sequence(_) => SequenceAction::GROUP,
+            Action::Transfer(_) => TransferAction::GROUP,
+            Action::ValidatorUpdate(_) => ValidatorUpdate::GROUP,
+            Action::SudoAddressChange(_) => SudoAddressChangeAction::GROUP,
+            Action::IbcRelayerChange(_) => IbcRelayerChangeAction::GROUP,
+            Action::Ics20Withdrawal(_) => Ics20Withdrawal::GROUP,
+            Action::InitBridgeAccount(_) => InitBridgeAccountAction::GROUP,
+            Action::BridgeLock(_) => BridgeLockAction::GROUP,
+            Action::BridgeUnlock(_) => BridgeUnlockAction::GROUP,
+            Action::BridgeSudoChange(_) => BridgeSudoChangeAction::GROUP,
+            Action::FeeChange(_) => FeeChangeAction::GROUP,
+            Action::FeeAssetChange(_) => FeeAssetChangeAction::GROUP,
+            Action::Ibc(_) => IbcRelay::GROUP,
+            Action::IbcSudoChange(_) => IbcSudoChangeAction::GROUP,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub(super) enum ActionGroup {
+    General,
+    UnbundleableGeneral,
+    Sudo,
+    UnbundleableSudo,
+}
+
+impl ActionGroup {
+    pub(super) fn is_bundleable(self) -> bool {
+        matches!(self, ActionGroup::General | ActionGroup::Sudo)
+    }
+
+    pub(super) fn is_sudo(self) -> bool {
+        matches!(self, ActionGroup::Sudo)
+    }
+}
+
+impl fmt::Display for ActionGroup {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ActionGroup::General => write!(f, "general"),
```

Comment:
Just leaving a note that should you go with my bikeshedding to rename prefix with `Bundleable*`, to also change the display name.